app/src/main/java/com/example/compassofukraine/ui/screen/DetailedEventScreen.kt

```diff
@@ -0,0 +1,224 @@
+            onClick = {
+                event?.let {
+                    openMapWithDirections(
+                        context,
+                        it.coordinates.latitude,
+                        it.coordinates.longitude
+                    )
+                }
+            }
+        ) {
+            Text(text = stringResource(id = R.string.show_on_the_map))
+        }
+    }
+}
+
+@Composable
+fun DetailedEventSheet(eventDetail: DetailedEvent?) {
+    LazyColumn(modifier = Modifier.fillMaxSize()) {
+        eventDetail?.let {
+            item {
+                Row(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(horizontal = 16.dp),
+                    horizontalArrangement = Arrangement.SpaceBetween,
+                    verticalAlignment = Alignment.Top
+                ) {
+                    Column {
+                        Text(
+                            text = it.title,
+                            style = MaterialTheme.typography.headlineMedium,
+                            maxLines = 3,
+                            modifier = Modifier.fillMaxWidth(.5f)
+                        )
+                        Text(
+                            text = it.shortDescription,
+                            style = MaterialTheme.typography.labelMedium
+                        )
+                    }
+                    Column(horizontalAlignment = Alignment.End) {
+                        Text(
+                            text = "${it.dateStart.dayOfMonth}.${it.dateStart.month.value}.${it.dateStart.year}",
+                            style = MaterialTheme.typography.bodyLarge
+                        )
+                        Text(
+                            text = when (it.eventTime) {
+                                is EventTime.AllDay -> stringResource(id = R.string.all_day_label)
+                                is EventTime.Time ->
+                                    "${(it.eventTime as EventTime.Time).start} - ${(it.eventTime as EventTime.Time).finish}"
+                            },
+                            style = MaterialTheme.typography.labelMedium
+                        )
+                    }
+                }
+                Text(
+                    text = it.description,
+                    style = MaterialTheme.typography.bodyMedium,
+                    modifier = Modifier.padding(top = 24.dp, start = 16.dp, end = 16.dp)
+                )
+            }
+        } ?: run {
+            item {
+                Row(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .padding(horizontal = 16.dp),
+                    horizontalArrangement = Arrangement.SpaceBetween,
+                    verticalAlignment = Alignment.Top
+                ) {
+                    Column {
+                        Text(
+                            text = "",
```

Comment:
Instead of changing value to empty string in case when we load data we should show to the user some kind of mocked view in my mind the best decision is to show gray rectangle which shimmers in shades of gray, thereby showing that the download process is taking place